tests/test_async_client.py

```
@@ -0,0 +1,1722 @@
+        mock_session = mocker.MagicMock()
+        mock_session.closed = False
+        mock_session.close = AsyncMock()
+        client._session = mock_session
+
+        await client.close()
+
+        mock_session.close.assert_called_once()
+        assert client._session is None
+        assert client._closed is True
+
+    async def test_close_without_session(self, mocker):
```

Ruff (0.15.0)

[warning] Unused method argument: `mocker`

(ARG002)